Decred Wallet/Features/History/TransactionHistoryViewController.swift

```diff
@@ -19,25 +20,28 @@ class TransactionHistoryViewController: UIViewController {
         return refreshControl
     }()
     
-    @IBOutlet weak var syncLabel: UILabel!
     
+    @IBOutlet weak var syncLabel: UILabel!
     @IBOutlet var tableView: UITableView!
     @IBOutlet var btnFilter: DropMenuButton!
-    var FromMenu = true
-    var visible:Bool = false
     
-    var filterMenu = [LocalizedStrings.all] as [String]
-    var filtertitle = [0] as [Int]
+    var filterMenu = [LocalizedStrings.all]
+    var filters: [Int32] = [0]
     
-    var mainContens = [Transaction]()
-    var Filtercontent = [Transaction]()
+    var transactions = [Transaction]()
+    var filteredItems = [Transaction]() {
+        didSet {
+            self.tableView.reloadData()
+        }
+    }
+    
+    var filterActive: Bool = false
     
     override func viewDidLoad() {
         super.viewDidLoad()
         self.initFilterBtn()
         self.tableView.addSubview(self.refreshControl)
         self.tableView.register(UINib(nibName: TransactionTableViewCell.identifier, bundle: nil), forCellReuseIdentifier: TransactionTableViewCell.identifier)
-
```

Comment:
This empty line space is important IMO. Indicates separation of concern between view loading and background tx notification listening.

Reply:
resolved in 59b4782